### ISSUE TITLE
Support `DependsOn`

### ISSUE THREAD

**Issue by awolden** [horike37/serverless-step-functions#158]
# This is a Feature Proposal

## Description

This plugin should support the `dependsOn:` property when generating the cloudformation for a workflow.

The main use case is for when we depend on an manually defined state machine role in the serverless.yml file. With the new batch functionality in workflows we have to manually create our role, but there is no way to do:

```
dependsOn: StateMachineRole
role:
  Ref: StateMachineRole
```

This results in the cloudformation failing because the role isn't created by the time the workflow is created.

Thanks,

-Alex

**Comment by horike37** [comment]
@awolden 
Thank you for your proposal 👍 
That sounds good 💯 I just added help wanted tag to this issue so that someone can pick it up.

**Comment by ldadams** [comment]
> # This is a Feature Proposal
> ## Description
> This plugin should support the `dependsOn:` property when generating the cloudformation for a workflow.
> 
> The main use case is for when we depend on an manually defined state machine role in the serverless.yml file. With the new batch functionality in workflows we have to manually create our role, but there is no way to do:
> 
> ```
> dependsOn: StateMachineRole
> role:
>   Ref: StateMachineRole
> ```
> This results in the cloudformation failing because the role isn't created by the time the workflow is created.
> 
> Thanks,
> 
> -Alex

Hey @awolden  - Do you have a work around for this?  I am having issues getting step functions deployed with batch integrations.  The deploy runs for a very long time with an internal error.  My lambdas get deployed but seems to fail creating the state machine.  Thanks for any help.

**Comment by awolden** [comment]
I made the role in a separate cloud-formation template and checked it in with the repo. You only really have to deploy/update it once.  Ended up looking like this:

```json
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "Generic State Machine Role",
  "Resources": {
    "StateMachineRole": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Principal": {
                "Service": [
                  "states.amazonaws.com"
                ]
              },
              "Action": [
                "sts:AssumeRole"
              ]
            }
          ]
        },
        "Path": "/",
        "RoleName": "dpc-generic-stateMachineRole",
        "Policies": [
          {
            "PolicyName": "batchJobManagement",
            "PolicyDocument": {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "batch:SubmitJob",
                    "batch:DescribeJobs",
                    "batch:TerminateJob"
                  ],
                  "Resource": "*"
                }
              ]
            }
          },
          {
            "PolicyName": "allEventsForBatch",
            "PolicyDocument": {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "events:PutTargets",
                    "events:PutRule",
                    "events:DescribeRule"
                  ],
                  "Resource": {
                    "Fn::Sub": "arn:aws:events:${AWS::Region}:${AWS::AccountId}:rule/StepFunctionsGetEventsForBatchJobsRule"
                  }
                }
              ]
            }
          },
          {
            "PolicyName": "executeLambda",
            "PolicyDocument": {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "lambda:InvokeFunction"
                  ],
                  "Resource": "*"
                }
              ]
            }
          }
        ]
      }
    }
  }
}
```

**Comment by ldadams** [comment]
Thank you - This was very helpful.

**Comment by theburningmonk** [comment]
@awolden @ldadams if you update to the latest version, the plugin should be generating the permissions for batch already, see #161 for more details

**Comment by horike37** [comment]
This has been shipped in v1.11.0 🎉
